src/libraries/Common/src/Interop/Windows/SspiCli/SecuritySafeHandles.cs

```diff
@@ -431,34 +432,40 @@ internal abstract partial class SafeDeleteContext : SafeHandle
             SafeFreeContextBuffer outFreeContextBuffer = null;
             try
             {
-                Span<Interop.SspiCli.SecBuffer> inUnmanagedBuffer = stackalloc Interop.SspiCli.SecBuffer[inSecurityBufferDescriptor.cBuffers];
-                inUnmanagedBuffer.Clear();
+                Span<Interop.SspiCli.SecBuffer> inUnmanagedBuffer = stackalloc Interop.SspiCli.SecBuffer[3];
+                for (int index = 0; index < inSecBuffers.Count; ++index)
```

Comment:
Can we add an assert here about the relationship between inSecBuffers.Count and the hardcoded 3 limit?

Reply:
there is `Debug.Assert(inSecBuffers.Count <= 3);` on L416